Use webpack-dev-server for development

Alexendoo opened the pull request:
Simplifies development somewhat - you just need to npm start and it will build automatically and update the bundle if the relevant files are modified. From there it can be opened in the browser as normal

The `process.env.PORT` configuration is kept around for convenience, though a user with the server installed and in their path can override options on the CLI themselves, for instance to enable browser auto-refresh

prawnsalad commented:
I'm trying to get it to show the development URL after all it's building output, ideas? Otherwise the building kicks in and the dev URL is pushed off the screen without time to notice it.

prawnsalad commented:
Oh, also, lets add `host: process.env.HOST || '127.0.0.1'` to the config to go with the port option.

Alexendoo commented:
Added noInfo which only shows output if it hits an error, also added the `process.env.HOST` bit and inline because otherwise it was defaulting to an iframe based URL, this will mean that it automatically refreshes the browser when you make changes though
